concert/app/src/functions/on_event_stage_changed/index.py

```diff
@@ -88,3 +104,107 @@ def get_song_list_update_transaction_item(message_data):
     }
 
     return {"Update": song_list_update_args}
+
+
+def get_song_list_get_transaction_item():
+    serializer = boto3.dynamodb.types.TypeSerializer()
+    song_list_get_args = {
+        "Key": {ATTR_RECORD_ID: serializer.serialize(RECORD_ID_SONG_LIST)},
+        "TableName": DYNAMODB_TABLE_NAME,
+    }
+
+    return {"Get": song_list_get_args}
+
+
+def get_aggregate_data_get_transaction_item():
+    serializer = boto3.dynamodb.types.TypeSerializer()
+    aggregate_data_get_args = {
+        "Key": {ATTR_RECORD_ID: serializer.serialize(RECORD_ID_AGGREGATE)},
+        "TableName": DYNAMODB_TABLE_NAME,
+    }
+
+    return {"Get": aggregate_data_get_args}
+
+
+def get_song_list_and_aggregate_data():
+    song_list_get_transaction_item = get_song_list_get_transaction_item()
+    aggregate_data_get_transaction_item = get_aggregate_data_get_transaction_item()
+
+    transaction_items = [
+        song_list_get_transaction_item,
+        aggregate_data_get_transaction_item,
+    ]
+
+    responses = dynamodb.transact_get_items(TransactItems=transaction_items)[
+        "Responses"
+    ]
+
+    deserializer = boto3.dynamodb.types.TypeDeserializer()
+    songs = {k: deserializer.deserialize(v) for k, v in responses[0]["Item"].items()}
+    aggregate_data = {
+        k: deserializer.deserialize(v) for k, v in responses[1]["Item"].items()
+    }
+
+    return songs, aggregate_data
+
+
+def get_songs_with_aggregate_choices(song_list, aggregate_data):
+    songs_with_aggregate_choices = []
+    for song_metadata in song_list[ATTR_SONG_LIST_SONGS]:
+        display_name = song_metadata[ATTR_SONG_LIST_DISPLAY_NAME]
+        start_time = song_metadata[ATTR_SONG_LIST_START_TIME]
+        end_time = song_metadata[ATTR_SONG_LIST_END_TIME]
+
+        choice_type = None
+
+        for timestamp in aggregate_data[ATTR_CHOICE_VALUE_COLOR].keys():
+            if start_time <= timestamp < end_time:
```

Comment:
For this data comparison and the others below, we probably want to parse to `datetime` rather than doing string comparisons. Agreed?

Reply:
If they're ISO format, it doesn't matter.